ui/src/app/views/resource-card/component.tsx

```diff
@@ -15,6 +20,84 @@ import {environment} from 'environments/environment';
 import {ConfirmDeleteModalComponent} from 'app/views/confirm-delete-modal/component';
 import {EditModalComponent} from 'app/views/edit-modal/component';
 
+const MenuItem = ({ icon, children, ...props }) => {
+  return <Clickable
+    {...props}
+    data-test-id={icon}
+    style={{
+      display: 'flex', alignItems: 'center',
+      minWidth: '5rem', height: '1.3333rem',
+      padding: '0 1rem', color: '#4A4A4A'
+     }}
+     hover={{backgroundColor: '#E0EAF1'}}
+  ><ClrIcon shape={icon} />&nbsp;{children}</Clickable>;
+};
+
+const ResourceCardMenu: React.FunctionComponent<{
+  disabled: boolean, resourceType: ResourceType, onRenameNotebook: Function,
+  onCloneResource: Function, onDeleteResource: Function, onEditCohort: Function,
+  onReviewCohort: Function, onEditConceptSet: Function
+}> = ({
+  disabled, resourceType, onRenameNotebook, onCloneResource,
+  onDeleteResource, onEditCohort, onReviewCohort, onEditConceptSet
+}) => {
+  return <PopupTrigger
+    side='bottom'
+    closeOnClick
+    content={
+      switchCase(resourceType,
+        ['notebook', () => {
+          return <React.Fragment>
+            <MenuItem icon='pencil' onClick={onRenameNotebook}>Rename</MenuItem>
+            <MenuItem icon='copy' onClick={onCloneResource}>Clone</MenuItem>
+            <MenuItem icon='trash' onClick={onDeleteResource}>Delete</MenuItem>
+          </React.Fragment>;
+        }],
+        ['cohort', () => {
+          return <React.Fragment>
+            <MenuItem icon='copy' onClick={onCloneResource}>Clone</MenuItem>
+            <MenuItem icon='pencil' onClick={onEditCohort}>Edit</MenuItem>
+            <MenuItem icon='grid-view' onClick={onReviewCohort}>Review</MenuItem>
+            <MenuItem icon='trash' onClick={onDeleteResource}>Delete</MenuItem>
+          </React.Fragment>;
+        }],
+        ['conceptSet', () => {
+          return <React.Fragment>
+            <MenuItem icon='pencil' onClick={onEditConceptSet}>Edit</MenuItem>
+            <MenuItem icon='trash' onClick={onDeleteResource}>Delete</MenuItem>
+          </React.Fragment>;
+        }]
+      )
+    }
+  >
+    <Clickable disabled={disabled} data-test-id='resource-menu'>
```

Comment:
This `data-test-id` is a technique for identifying specific elements for testing purposes. Using a data attribute makes its meaning clear and avoids overloading CSS classes for that purpose.